joblib/externals/loky/process_executor.py

```diff
@@ -180,6 +180,9 @@ def flag_as_broken(self, broken):
 _global_shutdown = False
 
 
+_debug_exit_lock_temp = threading.Lock()
```

Comment:
Do we need this? ^^

Reply:
Yes, but maybe in a different style or at least rename. And, it's only necessary with pypy.

The `PypyGeneratorExitThread` can execute `shutdown` concurrently with the `_python_exit` atexit call, which raises errors because the `ExecutorManagerThread` is joined two times.

Currently it doesn't show in unit tests. But you're likely to see the relevant traceback in the warnings if you run only the `test_multiple_generator_call_separated_gc` unit test.